test/test_svaerd_kalisch_1d.jl

```diff
@@ -46,6 +69,20 @@ end
     @test_allocations(semi, sol, allocs=750_000)
 end
 
+@testitem "svaerd_kalisch_1d_dingemans_fourier" setup=[Setup, SvaerdKalischEquations1D] begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                 "svaerd_kalisch_1d_dingemans_fourier.jl"),
+                        tspan=(0.0, 1.0),
+                        l2=[0.22799246923254327 0.7520172891302948 0.0],
+                        linf=[0.03671494177483947 0.12129171577180138 0.0],
+                        cons_error=[8.526512829121202e-14 5.3078570574495334e-5 0.0],
+                        change_waterheight=-8.526512829121202e-14,
+                        change_entropy=-0.0002424441479433881,
+                        change_entropy_modified=-4.00007138523506e-9)
+
+    @test_allocations(semi, sol, allocs=13_000_000)
```

Comment:
Is it expected that the Fourier operator allocates so much for the Svärd-Kalisch equations?

Reply:
It likely creates a lot of dense matrices for the linear solves, doesn't it?

Reply:
Yes, it needs to allocate a lot of dense matrices. I mainly just wanted to point out that here we need a lot more allocations than in the other tests. But there is probably nothing to do against it. If you agree we can just continue, I think.

Reply:
Yes, let's go ahead